src/analyses/goto_check.cpp

```diff
@@ -1666,6 +1666,34 @@ void goto_checkt::goto_check(
         }
       }
     }
+    else if(i.is_decl())
+    {
+      if(enable_pointer_check)
```

Comment:
I am unfamiliar with this code so non-blocking (I don't understand why dead statements need this instrumentation), but some documentation of why a deceleration statement needs to be instrumented when `enable_pointer_check` is on would be appreciated